haystack/nodes/reader/table.py

```diff
@@ -116,68 +137,164 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
+            context=document.content,
+            offsets_in_document=answer_offsets,
+            offsets_in_context=answer_offsets,
+            document_id=document.id,
+            meta={"aggregation_operator": current_aggregation_operator,
+                  "answer_cells": current_answer_cells}
+        )
+
+        return answer
+
+    def _predict_tapas_for_scored_qa(self, inputs: BatchEncoding, document: Document) -> List[Answer]:
+        table: pd.DataFrame = document.content
+
+        # Forward pass through model
+        outputs = self.model.tapas(**inputs)
+
+        # Get general table score
+        table_score = self.model.classifier(outputs.pooler_output)
+        table_score = table_score[0][1] - table_score[0][0]
+
+        # Get possible answer spans
+        token_types = [
+            "segment_ids",
+            "column_ids",
+            "row_ids",
+            "prev_labels",
+            "column_ranks",
+            "inv_column_ranks",
+            "numeric_relations",
+        ]
+        row_ids: List[int] = inputs.token_type_ids[:, :, token_types.index("row_ids")].tolist()[0]
+        column_ids: List[int] = inputs.token_type_ids[:, :, token_types.index("column_ids")].tolist()[0]
+
+        possible_answer_spans: List[Tuple[int, int, int, int]] = []  # List of tuples: (row_idx, col_idx, start_token, end_token)
+        current_start_idx = -1
+        current_column_id = -1
+        for idx, (row_id, column_id) in enumerate(zip(row_ids, column_ids)):
+            if row_id == 0 or column_id == 0:
+                continue
+            # Beginning of new cell
+            if column_id != current_column_id:
+                if current_start_idx != -1:
+                    possible_answer_spans.append(
+                        (row_ids[current_start_idx]-1, column_ids[current_start_idx]-1, current_start_idx, idx-1)
+                    )
+                current_start_idx = idx
+                current_column_id = column_id
+        possible_answer_spans.append(
+            (row_ids[current_start_idx]-1, column_ids[current_start_idx]-1, current_start_idx, len(row_ids)-1)
+        )
+
+        # Concat logits of start token and end token of possible answer spans
+        sequence_output = outputs.last_hidden_state
+        concatenated_logits = []
+        for possible_span in possible_answer_spans:
+            start_token_logits = sequence_output[0, possible_span[2], :]
+            end_token_logits = sequence_output[0, possible_span[3], :]
+            concatenated_logits.append(torch.cat((start_token_logits, end_token_logits)))
+        concatenated_logit_tensors = torch.unsqueeze(torch.stack(concatenated_logits), dim=0)
+
+        # Calculate score for each possible span
+        span_logits = torch.einsum("bsj,j->bs", concatenated_logit_tensors, self.model.span_output_weights) \
+                      + self.model.span_output_bias
+        span_logits_softmax = torch.nn.functional.softmax(span_logits, dim=1)
```

Comment:
Yes, Bogdan and I discussed that topic. We agreed (correct me if I am wrong @bogdankostic) that the table_score is more import for ranking than the span_score. Unfortunately the scores of spans from different tables cannot be compared. To prevent multiple answers from having the exact same score (if we took the table_score only), we want to show the span_score to the user in addition to the table_score.  
The scenario that you describe with a high score for a "bad table" that has a single "decent answer candidate" but no other good candidates should be ranked low based on the table_score.
If we combine table_score and span_score, we would need to ensure that sorting by the combined score, the results are first sorted by table_score and then by span_score. No matter how large the span_score of a result, the result should be ranked lower than any other result with a larger table_score.

Reply:
Yeah, I think the sorting is alright, but imagine TableQA in our current demo UI (or the DC Search UI). As a user, you will see the answers one after the other. Each of them has a single "relevance score" in the UI. What would you display there right now? The current `Answer.score` would be misleading here as score(result_rank_1) < score(result_rank_2).

How about adding both individual scores to meta and using a combined one in `Answer.score`? 
Something in the direction of `score = expit( (table_score * 100 + span_score) / 100)` (calibrating the constants there to something that results in a meaningful range between 0-1)